File: Solution/每日一题/2024.9.3/3.java
class Solution {
    public long maxStrength(int[] nums) {
        long max_num=nums[0];
        long min_num=nums[0];
        for(int i=1;i<nums.length;i++){
            int x=nums[i];
            long temp=max_num;
            max_num=Math.max(Math.max(max_num,x),Math.max(max_num*x,min_num*x));
            min_num=Math.min(Math.min(min_num,x),Math.min(temp*x,min_num*x));
        }
        return max_num;
    }
}

File: Solution/每日一题/2024.9.3/3.cpp
class Solution {
public:
    long long maxStrength(vector<int>& nums) {
        long long mn = nums[0], mx = mn;
        for (int i = 1; i < nums.size(); i++) {
            long long x = nums[i];
            long long tmp = mn;
            mn = min({mn, x, mn * x, mx * x});
            mx = max({mx, x, tmp * x, mx * x});
        }
        return mx;
    }
};

